app/models/concerns/audit_extensions.rb

```diff
@@ -65,8 +65,8 @@ def fix_auditable_type
     # STI Host class should use the stub module instead of Host::Base
     self.auditable_type = "Host"          if auditable_type =~  /Host::/
     self.associated_type = "Host"         if associated_type =~ /Host::/
-    self.auditable_type = auditable.type  if auditable_type == "Taxonomy" && auditable
-    self.associated_type = auditable.type if auditable_type == "Taxonomy" && auditable
+    self.auditable_type = auditable.type if auditable && ["Taxonomy", "LookupKey"].include?(auditable_type)
+    self.associated_type = auditable.type if auditable && ["Taxonomy", "LookupKey"].include?(auditable_type)
```

Comment:
This line should be changed anyway for: http://projects.theforeman.org/issues/19354

One thing I could think of that could break is the callbacks we have today on `before_save` that might have counted on `auditable` being accessed in this callback.
I moved the callback and did a few create/update/delete that showed up correctly in the audits and didn't see anything break right away.